llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -3916,12 +3916,12 @@ static Value *simplifyICmpInst(unsigned Predicate, Value *LHS, Value *RHS,
           // is non-negative then LHS <s RHS.
           case ICmpInst::ICMP_SGT:
           case ICmpInst::ICMP_SGE:
-            return ConstantExpr::getICmp(ICmpInst::ICMP_SLT, C,
-                                         Constant::getNullValue(C->getType()));
+            return ConstantFoldCompareInstruction(
```

Comment:
Use ConstantFoldCompareInstOperands instead, which is DataLayout-aware.

Reply:
Does that matter here? `C` came from m_ImmConstant and we're comparing it to null.

Reply:
It makes no functional difference here, but you should never use the functions declared in llvm/IR/ConstantFold.h, only those in llvm/Analysis/ConstantFolding.h, which is the public API. We should maybe move the former into the `detail` namespace to make this clearer.

Reply:
Or possibly we can move the ConstantFolder implementation out-of-line, because that's the only reason ConstantFold.h is a public header at all.